unity-renderer/Assets/Scripts/MainScripts/DCL/Analytics/SocialAnalytics/SocialAnalytics.cs

```diff
@@ -0,0 +1,270 @@
+            data.Add("to", toPlayerType.ToString());
+
+            GenericAnalytics.SendAnalytic(PLAYER_UNMUTED, data);
+        }
+
+        public void SendVoiceMessageStartedByFirstTime() { GenericAnalytics.SendAnalytic(VOICE_MESSAGE_STARTED_BY_FIRST_TIME); }
+
+        public void SendVoiceMessageSent(double messageLength)
+        {
+            Dictionary<string, string> data = new Dictionary<string, string>();
+            data.Add("length", messageLength.ToString());
+
+            GenericAnalytics.SendAnalytic(VOICE_MESSAGE_SENT, data);
+        }
+
+        public void SendChannelMessageSent(string fromUserId, double messageLength, string channel)
+        {
+            PlayerType? fromPlayerType = GetPlayerTypeByUserId(fromUserId);
+
+            Dictionary<string, string> data = new Dictionary<string, string>();
+            data.Add("from", fromPlayerType.ToString());
+            data.Add("length", messageLength.ToString());
+            data.Add("channel", channel);
+
+            GenericAnalytics.SendAnalytic(CHANNEL_MESSAGE_SENT, data);
+        }
+
+        public void SendChannelMessageReceived(string fromUserId, double messageLength, string channel, ChatMessage.Type messageType)
+        {
+            PlayerType? fromPlayerType = GetPlayerTypeByUserId(fromUserId);
+
+            Dictionary<string, string> data = new Dictionary<string, string>();
+            data.Add("from", fromPlayerType.ToString());
+            data.Add("length", messageLength.ToString());
+            data.Add("channel", channel);
+            data.Add("message_type", messageType.ToString());
+
+            GenericAnalytics.SendAnalytic(CHANNEL_MESSAGE_RECEIVED, data);
+        }
+
+        public void SendDirectMessageSent(string fromUserId, string toUserId, double messageLength, bool areFriends, ChatContentType contentType)
+        {
+            PlayerType? fromPlayerType = GetPlayerTypeByUserId(fromUserId);
+            PlayerType? toPlayerType = GetPlayerTypeByUserId(toUserId);
+
+            Dictionary<string, string> data = new Dictionary<string, string>();
+            data.Add("from", fromPlayerType.ToString());
+            data.Add("to", toPlayerType.ToString());
+            data.Add("length", messageLength.ToString());
+            data.Add("friends", areFriends.ToString());
+            data.Add("content_type", contentType.ToString());
+
+            GenericAnalytics.SendAnalytic(DIRECT_MESSAGE_SENT, data);
+        }
+
+        public void SendDirectMessageReceived(string fromUserId, string toUserId, double messageLength, bool areFriends, ChatContentType contentType)
+        {
+            PlayerType? fromPlayerType = GetPlayerTypeByUserId(fromUserId);
+            PlayerType? toPlayerType = GetPlayerTypeByUserId(toUserId);
+
+            Dictionary<string, string> data = new Dictionary<string, string>();
+            data.Add("from", fromPlayerType.ToString());
```

Comment:
In case of `fromPlayerType` or `toPlayerType` is null, we shouldn't send the analytic. I missed do this check. Done in https://github.com/decentraland/unity-renderer/pull/2257/commits/678828e6b8345a710e289d0af3ae46fb12273a26

Reply:
This is intentional; if `GetPlayerTypeByUserId()` return null, then `fromPlayerType` (or `toPlayerType`) will be assigned as null and we want to send this info as so in the metric.